layouts/blog/post.html

```diff
@@ -9,8 +9,24 @@ <h1><a href="{{.Page.Permalink}}">{{ .Title }}</a></h1>
     <div class="post">
         <aside>
             <div class="summary">{{ .Params.preview }}</div>
-            <time itemprop="published" datetime="{{ dateFormat "2006-01-02" .PublishDate }}" title="{{ dateFormat "Monday, January 2, 2006" .PublishDate }}">{{ dateFormat "January 2, 2006" .PublishDate }}</time>
-            <p>by <span rel="author">{{ .Params.author }}</span></p>
+            <div class="info">
+                {{ if .Params.github_handle }}
+                    <div class="post-avatar">
+                        {{ $urlPre := "https://api.github.com" }}
+                        {{ $gist := getJSON $urlPre "/users/" .Params.github_handle }}
+                        {{ with $gist }}
+                            <img src="{{ .avatar_url }}" alt="Avatar" class="avatar">
+                        {{ end }}
+                    </div>
+                <div class="date-and-name">
+                    <p>by <span rel="author">{{ .Params.author }}</span></p>
+                    <time itemprop="published" datetime="{{ dateFormat "2006-01-02" .PublishDate }}" title="{{ dateFormat "Monday, January 2, 2006" .PublishDate }}">{{ dateFormat "January 2, 2006" .PublishDate }}</time>
+                </div>
+                {{ else }}
+                    <time itemprop="published" datetime="{{ dateFormat "2006-01-02" .PublishDate }}" title="{{ dateFormat "Monday, January 2, 2006" .PublishDate }}">{{ dateFormat "January 2, 2006" .PublishDate }}</time>
+                    <p>by <span rel="author">{{ .Params.author }}</span></p>
+                {{ end }}
```

Comment:
These two blocks are almost identical, don't duplicate markup makes it more difficult to maintain. No need to reorder elements with modern CSS such as flex layout